Bootloaders/DFU/BootloaderDFU.c

```diff
@@ -136,8 +136,8 @@ void Application_Jump_Check(void)
 		if (!(BootloaderAPI_ReadFuse(GET_HIGH_FUSE_BITS) & ~FUSE_BOOTRST))
 		{
 			/* If the reset source was not an external reset or the key is correct, clear it and jump to the application */
-			//if (!(MCUSR & (1 << EXTRF)) || (MagicBootKey == MAGIC_BOOT_KEY))
-			//  JumpToApplication = true;
+			if (!(MCUSR & (1 << EXTRF)) || (MagicBootKey == MAGIC_BOOT_KEY))
```

Comment:
What is this for again?

Reply:
without it, the bootloader never jumps to the application code.  Eg, it gets stuck in bootloader.  `dfu-programer reset` can be run again and again and again. 

Uncommenting that fixes that, and brings it back in line with upstream.